Source/WebKit/UIProcess/mac/WebViewImpl.mm

```diff
@@ -4697,6 +4697,31 @@ static NSPasteboardName pasteboardNameForAccessCategory(WebCore::DOMPasteAccessC
     return validAttributes.get().get();
 }
 
+static Vector<WebCore::CompositionUnderline> extractUnderlines(NSAttributedString *string)
+{
+    Vector<WebCore::CompositionUnderline> result;
```

Comment:
do we know what this function does that `compositionUnderlines` doesn't do that causes the broken behavior?

Reply:
`compositionUnderlines` doesn't populate the result for dictated marked text, at least it didn't prior to the update in this PR.

Reply:
> compositionUnderlines doesn't populate the result for dictated marked text

Do we know why?

Reply:
Yes, at some point it resulted in an extra blue underline under the dictated text. Not sure if that is still the case, but any change is outside the scope of an upstreaming PR.

Reply:
Yes confirmed the underlines still incorrectly appear if I add this:
```
diff --git a/Source/WebKit/UIProcess/mac/WebViewImpl.mm b/Source/WebKit/UIProcess/mac/WebViewImpl.mm
index 94951f5b2259..5e453be761fb 100644
--- a/Source/WebKit/UIProcess/mac/WebViewImpl.mm
+++ b/Source/WebKit/UIProcess/mac/WebViewImpl.mm
@@ -5042,8 +5042,8 @@ static Vector<WebCore::CompositionUnderline> compositionUnderlines(NSAttributedS
 
         NSColor *underlineColor = attributes[NSUnderlineColorAttributeName];
         bool isClear = [underlineColor isEqual:NSColor.clearColor];
-
-        if (!isClear)
+        (void)isClear;
+        // if (!isClear)
```